### DISCUSSION TITLE
Timetable for fiscal year

### DISCUSSION THREAD

**Discussion by Robindeep-Kaur** [apache/airflow#25893] [Q&A] [answered]
I've been trying to create a fiscal year/accounting year calendar for DAG as a timetable plugin, so that the dag can run on random dates of a month for a year. I wanted to ask if there's any implementation of the same is available.

Or at least an implementation of scheduled DAG runs for random dates of a month.
Example: 2, 4, 25 for January, 3, 16, 22 for February and so on.

Here's a link to an example of what a fiscal year is: https://www.calendarpedia.com/fiscal-calendar-2022-pdf-templates.html

**Comment by potiuk** [comment]
You can see all the community timetables in the source code. We have no other implementation in the community at least - maybe someone  has other implementations but it's not indexed or known to us.

**Reply by Robindeep-Kaur** [reply]
I just came across this AIP, as this is something that I've been trying to realise:
https://cwiki.apache.org/confluence/display/AIRFLOW/%5BWIP%5D+AIP-55+Rule-based+timetable+with+logical+composition
I wanted to know how far along the discussion is. I don't see any recent updates on it.

**Reply by potiuk** [reply, marked as answer]
That's it. It stalled and author did not follow throuhg - but you might pick it up and lead to completion if you want.

**Reply by Robindeep-Kaur** [reply]
Makes sense. Maybe I'll get into it. Thanks!

**Comment by bambooj** [comment]
邮件已经收到...谢谢...
